openeo/rest/imagecollectionclient.py

```diff
@@ -970,8 +970,29 @@ def save_result(self, format: str = "GTIFF", options: dict = None):
             }
         )
 
-    def download(self, outputfile: str = None, format: str = "GTIFF", options: dict = None):
+    def download(self, outputfile: str = None, format: str = None, options: dict = None):
         """Download image collection, e.g. as GeoTIFF."""
+        if format == None:
+            dots = [i for i, ltr in enumerate(outputfile) if ltr == "."]
+            format_string = outputfile[dots[len(dots)-1]+1:len(outputfile)].upper()
+
+            if format_string == "TIFF":
+                format = "GTIFF"
+            elif format_string == "TIF":
+                format = "GTIFF"
+            elif format_string == "JSON":
+                format = "JSON"
+            elif format_string == "NC":
+                format = "NetCDF"
+            elif format_string == "CSV":
+                format = "CSV"
+            elif format_string == "PNG":
+                format = "PNG"
+            elif format_string == "COVJSON":
+                format = "COVJSON"
+            else:
+                raise AttributeError("A correct format string should be provided, as the format couldn't be detected from the outputfile.")
```

Comment:
`AttributeError` is not the correct exception to raise here (it's for when you have an object and request an attribute that doesn't exist).  A `ValueError` or `OpenEoClientException` would be better.

Reply:
Also, instead of raising an exception, we could also just use the file extension directly as format. If that doesn't make sense, the back-end will complain anyway